core/common/src/main/java/alluxio/util/ShellUtils.java

```diff
@@ -234,14 +330,113 @@ public String toString() {
   }
 
   /**
-   * Static method to execute a shell command.
+   * Object representation of a command execution.
+   */
+  public static class CommandReturn {
+    private int mExitCode;
+    private String mStdOut;
+    private String mStdErr;
+
+    /**
+     * Creates object from the contents.
+     *
+     * @param code exit code
+     * @param stdOut stdout content
+     * @param stdErr stderr content
+     */
+    public CommandReturn(int code, String stdOut, String stdErr) {
+      mExitCode = code;
+      mStdOut = stdOut;
+      mStdErr = stdErr;
+    }
+
+    /**
+     * Gets the stdout content.
+     *
+     * @return stdout content
+     */
+    public String getStdOut() {
+      return mStdOut;
+    }
+
+    /**
+     * Gets the stderr content.
+     *
+     * @return stderr content
+     */
+    public String getStdErr() {
+      return mStdErr;
+    }
+
+    /**
+     * Gets the exit code.
+     *
+     * @return exit code of execution
+     */
+    public int getExitCode() {
+      return mExitCode;
+    }
+
+    /**
+     * Formats the object to more readable format.
+     * This is not done in toString() because stdout and stderr may be long.
+     *
+     * @return pretty formatted output
+     */
+    public String getFormattedOutput() {
```

Comment:
Not overriding `toString()` cuz the stdout and stderr may have a good amount of info. I don't want people to unconsciously execute a lot of commands and print everything.